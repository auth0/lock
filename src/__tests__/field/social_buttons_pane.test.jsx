import React from 'react';
import { mount } from 'enzyme';

import { expectComponent, extractPropsFromWrapper, mockComponent } from 'testUtils';

jest.mock('ui/button/auth_button', () => mockComponent('auth_button'));

const getComponent = () => require('field/social/social_buttons_pane').default;

describe('SocialButtonsPane', () => {
  const defaultProps = {
    lock: {},
    bigButtons: false,
    labelFn: (...keys) => keys.join(','),
    showLoading: false,
    signUp: false
  };
  beforeEach(() => {
    jest.resetModules();

    jest.mock('quick-auth/actions', () => ({
      logIn: jest.fn()
    }));

    jest.mock('connection/social/index', () => ({
      displayName: () => 'displayName',
      socialConnections: () => [
        { item: 1, get: (key) => `socialConnections1-${key}` },
        { item: 2, get: (key) => `socialConnections2-${key}` }
      ],
      authButtonsTheme: () => ({
        get: () => ({
          get: (key) => `authButtonsTheme-${key}`
        })
      })
    }));

    jest.mock('core/index', () => ({
      id: () => 1
    }));
  });

  it('renders correctly', () => {
    const SocialButtonsPane = getComponent();
    expectComponent(
      <SocialButtonsPane
        {...defaultProps}
        />
    ).toMatchSnapshot();
  });
  it('renders big buttons when bigButtons === true', () => {
    const SocialButtonsPane = getComponent();
    expectComponent(
      <SocialButtonsPane
        {...defaultProps}
        bigButtons
        />
    ).toMatchSnapshot();
  });
  it('shows loading when showLoading === true', () => {
    const SocialButtonsPane = getComponent();
    expectComponent(
      <SocialButtonsPane
        {...defaultProps}
        showLoading
        />
    ).toMatchSnapshot();
  });
  it('shows header when instructions are available', () => {
    const SocialButtonsPane = getComponent();
    expectComponent(
      <SocialButtonsPane
        {...defaultProps}
        instructions="instructions"
        />
    ).toMatchSnapshot();
  });
  it('calls `logIn` with social connection 1 when first button is clicked', () => {
    let SocialButtonsPane = getComponent();

    const wrapper = mount(<SocialButtonsPane {...defaultProps} />);
    const props = extractPropsFromWrapper(wrapper, 2);

    props.onClick();

    const {mock} = require('quick-auth/actions').logIn;
    expect(mock.calls.length).toBe(1);
    expect(mock.calls[0]).toMatchSnapshot();
  });
  it('calls `logIn` with social connection 2 when second button is clicked', () => {
    let SocialButtonsPane = getComponent();

    const wrapper = mount(<SocialButtonsPane {...defaultProps} />);
    const props = extractPropsFromWrapper(wrapper, 3);

    props.onClick();

    const {mock} = require('quick-auth/actions').logIn;
    expect(mock.calls.length).toBe(1);
    expect(mock.calls[0]).toMatchSnapshot();
  });
});
