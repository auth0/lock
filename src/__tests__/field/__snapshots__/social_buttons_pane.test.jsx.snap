// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialButtonsPane calls \`logIn\` with social connection 1 when first button is clicked 1`] = `
Array [
  1,
  Object {
    "get": [Function],
    "item": 1,
  },
]
`;

exports[`SocialButtonsPane calls \`logIn\` with social connection 2 when second button is clicked 1`] = `
Array [
  1,
  Object {
    "get": [Function],
    "item": 2,
  },
]
`;

exports[`SocialButtonsPane disables social buttons when disabled === true 1`] = `
<div
  className="auth-lock-social-buttons-pane"
>
  <div
    className="auth0-lock-social-buttons-container"
  >
    <div
      data-__type="auth_button"
      data-disabled={true}
      data-foregroundColor="authButtonsTheme-foregroundColor"
      data-icon="authButtonsTheme-icon"
      data-isBig={false}
      data-label="loginWithLabel,authButtonsTheme-displayName"
      data-onClick={[Function]}
      data-primaryColor="authButtonsTheme-primaryColor"
      data-strategy="socialConnections1-strategy"
    />
    <div
      data-__type="auth_button"
      data-disabled={true}
      data-foregroundColor="authButtonsTheme-foregroundColor"
      data-icon="authButtonsTheme-icon"
      data-isBig={false}
      data-label="loginWithLabel,authButtonsTheme-displayName"
      data-onClick={[Function]}
      data-primaryColor="authButtonsTheme-primaryColor"
      data-strategy="socialConnections2-strategy"
    />
  </div>
</div>
`;

exports[`SocialButtonsPane renders big buttons when bigButtons === true 1`] = `
<div
  className="auth-lock-social-buttons-pane"
>
  <div
    className="auth0-lock-social-buttons-container"
  >
    <div
      data-__type="auth_button"
      data-disabled={false}
      data-foregroundColor="authButtonsTheme-foregroundColor"
      data-icon="authButtonsTheme-icon"
      data-isBig={true}
      data-label="loginWithLabel,authButtonsTheme-displayName"
      data-onClick={[Function]}
      data-primaryColor="authButtonsTheme-primaryColor"
      data-strategy="socialConnections1-strategy"
    />
    <div
      data-__type="auth_button"
      data-disabled={false}
      data-foregroundColor="authButtonsTheme-foregroundColor"
      data-icon="authButtonsTheme-icon"
      data-isBig={true}
      data-label="loginWithLabel,authButtonsTheme-displayName"
      data-onClick={[Function]}
      data-primaryColor="authButtonsTheme-primaryColor"
      data-strategy="socialConnections2-strategy"
    />
  </div>
</div>
`;

exports[`SocialButtonsPane renders correctly 1`] = `
<div
  className="auth-lock-social-buttons-pane"
>
  <div
    className="auth0-lock-social-buttons-container"
  >
    <div
      data-__type="auth_button"
      data-disabled={false}
      data-foregroundColor="authButtonsTheme-foregroundColor"
      data-icon="authButtonsTheme-icon"
      data-isBig={false}
      data-label="loginWithLabel,authButtonsTheme-displayName"
      data-onClick={[Function]}
      data-primaryColor="authButtonsTheme-primaryColor"
      data-strategy="socialConnections1-strategy"
    />
    <div
      data-__type="auth_button"
      data-disabled={false}
      data-foregroundColor="authButtonsTheme-foregroundColor"
      data-icon="authButtonsTheme-icon"
      data-isBig={false}
      data-label="loginWithLabel,authButtonsTheme-displayName"
      data-onClick={[Function]}
      data-primaryColor="authButtonsTheme-primaryColor"
      data-strategy="socialConnections2-strategy"
    />
  </div>
</div>
`;

exports[`SocialButtonsPane shows header when instructions are available 1`] = `
<div
  className="auth-lock-social-buttons-pane"
>
  <p>
    instructions
  </p>
  <div
    className="auth0-lock-social-buttons-container"
  >
    <div
      data-__type="auth_button"
      data-disabled={false}
      data-foregroundColor="authButtonsTheme-foregroundColor"
      data-icon="authButtonsTheme-icon"
      data-isBig={false}
      data-label="loginWithLabel,authButtonsTheme-displayName"
      data-onClick={[Function]}
      data-primaryColor="authButtonsTheme-primaryColor"
      data-strategy="socialConnections1-strategy"
    />
    <div
      data-__type="auth_button"
      data-disabled={false}
      data-foregroundColor="authButtonsTheme-foregroundColor"
      data-icon="authButtonsTheme-icon"
      data-isBig={false}
      data-label="loginWithLabel,authButtonsTheme-displayName"
      data-onClick={[Function]}
      data-primaryColor="authButtonsTheme-primaryColor"
      data-strategy="socialConnections2-strategy"
    />
  </div>
</div>
`;

exports[`SocialButtonsPane shows loading when showLoading === true 1`] = `
<div
  className="auth-lock-social-buttons-pane"
>
  <div
    className="auth0-lock-social-buttons-container"
  >
    <div
      data-__type="auth_button"
      data-disabled={false}
      data-foregroundColor="authButtonsTheme-foregroundColor"
      data-icon="authButtonsTheme-icon"
      data-isBig={false}
      data-label="loginWithLabel,authButtonsTheme-displayName"
      data-onClick={[Function]}
      data-primaryColor="authButtonsTheme-primaryColor"
      data-strategy="socialConnections1-strategy"
    />
    <div
      data-__type="auth_button"
      data-disabled={false}
      data-foregroundColor="authButtonsTheme-foregroundColor"
      data-icon="authButtonsTheme-icon"
      data-isBig={false}
      data-label="loginWithLabel,authButtonsTheme-displayName"
      data-onClick={[Function]}
      data-primaryColor="authButtonsTheme-primaryColor"
      data-strategy="socialConnections2-strategy"
    />
  </div>
  <div
    className="auth0-loading-container"
  >
    <div
      className="auth0-loading"
    />
  </div>
</div>
`;
